src/javaharness/java/arcs/api/Thing.java

```diff
@@ -0,0 +1,11 @@
+package arcs.api;
+
+public class Thing<T>  {
+    public final T thing;
+
+    Thing(T thing) {
+        this.thing = thing;
+    }
+
+    public T get() { return this.thing; }
```

Comment:
So the real pattern for this later will me to have multiple getters, like

public Optional<Handle> getHandle() { return this.thing instanceof Handle ? Optional.of(this.thing) : Optional.EMPTY }

Then when you fetch a thing, you get call the appropriate type-safe getter, and get back a non-null pointer to chain operations off of.

Reply:
i'll add these methods after https://github.com/PolymerLabs/arcs/pull/3187 is merged.